backstory/src/test/java/com/google/sps/servlets/BlobstoreManagerTest.java

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * Tests for the blobstore service wrapper, BlobstoreManager
+ */
+@RunWith(MockitoJUnitRunner.class)
+public final class BlobstoreManagerTest {
+  // All necessary mocks and objects for the test calls themselves
+  private BlobstoreService mockBlobstoreService;
+  private BlobstoreServiceConstantFields mockBlobstoreServiceConstantFields;
+  private BlobInfoFactory mockBlobInfoFactory;
+  private HttpServletRequest mockRequest;
+  private final String formInputElementName = "image-upload";
+  private final String mockBlobKeyString = "mockBlobKeyString";
+
+
+  /**
+   * Tests the getUploadedFileBlobKeyString method of BlobstoreManager
+   */
+  @Test
+  public void testGetUploadedFileBlobKeyString() throws IOException {
+    mockRequest = mock(HttpServletRequest.class);
+    mockBlobstoreService = mock(BlobstoreService.class);
+    mockBlobstoreServiceConstantFields = mock(BlobstoreServiceConstantFields.class);
+    mockBlobInfoFactory = mock(BlobInfoFactory.class);
+
+    BlobInfo mockBlobInfo = mock(BlobInfo.class);
+    Map<String, List<BlobKey>> mockInputToBlobkey = new HashMap<>();
+    List<BlobKey> mockBlobKeyStrings = new ArrayList<>();
+    mockBlobKeyStrings.add(new BlobKey(mockBlobKeyString));
+    mockInputToBlobkey.put(formInputElementName, mockBlobKeyStrings);
+
+    when(mockBlobstoreService.getUploads(any(HttpServletRequest.class)))
+        .thenReturn(mockInputToBlobkey);
+
+    // Mock one image getting uploaded
+    when(mockBlobInfo.getSize()).thenReturn(new Long(1));
+    when(mockBlobInfoFactory.loadBlobInfo(any(BlobKey.class)))
+        .thenReturn(mockBlobInfo);
+
+    BlobstoreManager blobstoreManager = new BlobstoreManager(mockBlobstoreService, mockBlobstoreServiceConstantFields, mockBlobInfoFactory);
+    Assert.assertEquals(mockBlobKeyString, blobstoreManager.getUploadedFileBlobKeyString(mockRequest, formInputElementName));
+  }
+
+  /**
+   * Tests the getBlobBytes method of BlobstoreManager
+   */
+  @Test
+  public void testGetBlobBytes() throws IOException {
+    mockRequest = mock(HttpServletRequest.class);
+    mockBlobstoreService = mock(BlobstoreService.class);
+    mockBlobstoreServiceConstantFields = mock(BlobstoreServiceConstantFields.class);
+    mockBlobInfoFactory = mock(BlobInfoFactory.class);
+
+    BlobInfo mockBlobInfo = mock(BlobInfo.class);
+    Map<String, List<BlobKey>> mockInputToBlobkey = new HashMap<>();
+    List<BlobKey> mockBlobKeyStrings = new ArrayList<>();
+    mockBlobKeyStrings.add(new BlobKey(mockBlobKeyString));
+    mockInputToBlobkey.put(formInputElementName, mockBlobKeyStrings);
+
+    when(mockBlobstoreService.getUploads(any(HttpServletRequest.class)))
+        .thenReturn(mockInputToBlobkey);
+
+    // Mock one image getting uploaded
+    when(mockBlobInfo.getSize()).thenReturn(new Long(1));
+    when(mockBlobInfoFactory.loadBlobInfo(any(BlobKey.class)))
+        .thenReturn(mockBlobInfo);
+
+    byte[] mockImageByteArray = new byte[0];
```

Comment:
Would suggest using a non-zero value here to provide confidence that the method is not accidentally returning 0.

Reply:
Sounds good, updated.